src/main/kotlin/bakalarka/elearningplatform/controller/ChapterController.kt

```diff
@@ -0,0 +1,30 @@
+package bakalarka.elearningplatform.controller
+
+import bakalarka.elearningplatform.model.CourseChapter
+import bakalarka.elearningplatform.request.AddCourseChapterRequest
+import bakalarka.elearningplatform.service.ChapterService
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class ChapterController(var chapterService: ChapterService) {
+
+    @PutMapping("/chapters/{chapterId}")
+    @PreAuthorize("hasAuthority('update:courses')")
+    fun updateChapter(
+        @PathVariable chapterId: Long,
+        @RequestBody request: AddCourseChapterRequest
+    ): ResponseEntity<CourseChapter> {
+        val response = chapterService.updateChapter(chapterId, request)
+        return if (response == null) {
+            ResponseEntity(HttpStatus.UNAUTHORIZED)
```

Comment:
Toto nie je moc pekny pattern - co keby service zacne vracat null aj ked je chyba niekde inde? Myslim, ze uz aj teraz vratis `HttpStatus.UNAUTHORIZED` aj ked chapter neexistuje.

Toto by mohlo vratit rovno `ResponseEntity.ok(chapterService.updateChapter(chapterId, request))` a service by mal hadzat `AuthorizationException` resp. `ResourceNotFoundException` ked nastanu dane chyby.

Ked sa to vyskytne este niekde v projekte, tak k tomu iba napisem ERROR_HANDLING poznamku, nech vies, ze aj to treba zmenit.